impeller/geometry/path.cc

```diff
@@ -246,11 +258,14 @@ Path::Polyline Path::CreatePolyline(
     const auto& component = components_[component_i];
     switch (component.type) {
       case ComponentType::kLinear:
-        return &linears_[component.index];
+        return reinterpret_cast<const LinearPathComponent*>(
```

Comment:
nit: you could use a union here instead of having these casts.

Reply:
A union would require us to align storage to the largest component size, effectively padding out linear components 4x. The cast approach means that we can store points inline/continuously, though we have to rely on the secondary vector of verbs to determine the correct offsets. This is approach is closer to what Skia uses and I think we should too (partially because it makes point/verb reserve work nicely), and partially because its closer to the format we need for compute anyway.

Reply:
Yea, could be.  I'm not familiar enough with the code to know if there are less items in the vector to account for the different type.  If the counts are the same then you're right.

You could make your own cast with template specialization 
```c++
case ComponentType::kLinear:
  return point_cast<ComponentType::kLinear>(&points_[component.index]);
```

That's a wee bit cleaner.  I don't have another easy solution.

Reply:
I'm going to punt on this for now.